src/lazy/text/buffer.rs

```diff
@@ -86,22 +86,6 @@ const WHITESPACE_CHARACTERS: &[char] = &[
 /// Same as [WHITESPACE_CHARACTERS], but formatted as a string for use in some `nom` APIs
 pub(crate) const WHITESPACE_CHARACTERS_AS_STR: &str = " \t\r\n\x09\x0B\x0C";
 
-/// This helper function takes a parser and returns a closure that performs the same parsing
-/// but prints the Result before returning the output. This is handy for debugging.
-// A better implementation would use a macro to auto-generate the label from the file name and
-// line number.
-fn dbg_parse<I: Debug, O: Debug, E: Debug, P: Parser<I, O, E>>(
-    label: &'static str,
-    mut parser: P,
-) -> impl Parser<I, O, E> {
-    move |input: I| {
-        let result = parser.parse(input);
-        #[cfg(debug_assertions)]
-        println!("{}: {:?}", label, result);
-        result
-    }
-}
-
```

Comment:
🗺️ This method would cause a clippy warning in release builds because `label` wouldn't be used. The text reader is spec compliant, so I feel we don't really need this debug helper anymore.